pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -69,11 +74,16 @@ def done(self) -> bool:
         max_steps_reached = self.max_steps is not None and self.global_step >= self.max_steps
         return max_steps_reached or self.trainer.should_stop or self._num_training_batches_reached(self.is_last_batch)
 
-    def connect(self, trainer: 'pl.Trainer', *args: Any, **kwargs: Any) -> None:
-        """Connects the loop with necessary arguments like the trainer"""
-        super().connect(trainer, *args, **kwargs)
-        self.batch_loop.connect(trainer)
-        self.val_loop.connect(trainer)
+    def connect(
+        self,
+        batch_loop: Optional[TrainingBatchLoop] = None,
+        val_loop: Optional["loops.EvaluationLoop"] = None,
+    ) -> None:
+        """Optionally connect a custom batch- or validation loop to this training epoch loop."""
```

Comment:
```suggestion
        """Optionally connect a custom batch or validation loop to this training epoch loop."""
```

Reply:
why? maybe it's just in german and not in english, but I wanted to abbreviate word repetation

batch loop or validation loop
to
batch- or validation loop

Reply:
Thought it was a typo.
Not sure if that's correct in English.

But AFAIK what you want is the difference between:

"batch or validation loop" -> "the batch loop or the validation loop"
vs
"batch, or validation loop" -> the batch or the validation loop"

Reply:
okay, let's keep it as an exercise for the reader
xD